examples/nffPktgen/generator/generator.go

```diff
@@ -229,6 +230,77 @@ func generateICMPIPv4(pkt *packet.Packet, config *PacketConfig, rnd *rand.Rand)
 	pktICMP.Cksum = packet.SwapBytesUint16(packet.CalculateIPv4ICMPChecksum(pktIP, pktICMP, pkt.Data))
 }
 
+func generatePcap(pkt *packet.Packet, config *PacketConfig, _ *rand.Rand) {
+	if pkt == nil {
+		panic("Failed to create new packet")
+	}
+	pcapConfig := &config.Pcap
+	if pcapConfig.FileReader == nil {
+		f, err := os.Open(pcapConfig.Path)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to open pcap file %s", pcapConfig.Path))
+		}
+		pcapConfig.FileReader = f
+		var glHdr packet.PcapGlobHdr
+		if err := packet.ReadPcapGlobalHdr(f, &glHdr); err != nil {
+			f.Close()
+			panic(fmt.Sprintf("Failed to read pcap file header, returned %v", err))
+		}
+	}
+	isEOF, err := pkt.ReadPcapOnePacket(pcapConfig.FileReader)
+	if err != nil {
+		panic("Failed to read one packet from pcap")
+	}
+	if isEOF {
+		if _, err := pcapConfig.FileReader.Seek(packet.PcapGlobHdrSize, 0); err != nil {
+			panic("Failed to parse pcap file header")
+		}
+		if _, err := pkt.ReadPcapOnePacket(pcapConfig.FileReader); err != nil {
+			panic("Failed to read one packet from pcap")
+		}
+	}
+}
+
+func generatePcapInMemory(pkt *packet.Packet, config *PacketConfig, _ *rand.Rand) {
+	if pkt == nil {
+		panic("Failed to create new packet")
+	}
+	pcapConfig := &config.Pcap
+	if !pcapConfig.ReachedEOF {
+		if pcapConfig.FileReader == nil {
+			f, err := os.Open(pcapConfig.Path)
+			if err != nil {
+				panic(fmt.Sprintf("Failed to open pcap file %s", pcapConfig.Path))
+			}
+			pcapConfig.FileReader = f
+			var glHdr packet.PcapGlobHdr
+			if err := packet.ReadPcapGlobalHdr(f, &glHdr); err != nil {
+				f.Close()
+				panic(fmt.Sprintf("Failed to read pcap file header, returned %v", err))
+			}
+		}
+		isEOF, err := pkt.ReadPcapOnePacket(pcapConfig.FileReader)
+		if err != nil {
+			panic("Failed to read one packet from pcap")
+		}
+		if isEOF {
+			pcapConfig.FileReader.Close()
+			pcapConfig.ReachedEOF = true
```

Comment:
In case of EOF, ReadPcapOnePacket doesn't initialize pkt. Setting pcapConfig.ReachedEOF to true here ensures that the code enters the condition at line 294 to get the packet to send from pcapConfig.Packets.